cpp/models/abm/world.cpp

```diff
@@ -85,49 +87,70 @@ void World::migration(TimePoint t, TimeSpan dt)
             m_enhanced_migration_rules.push_back(rule);
         }
     }
+
     for (auto& person : m_persons) {
         for (auto rule : m_enhanced_migration_rules) {
             //check if transition rule can be applied
-            auto target_type      = rule.first(*person, t, dt, m_migration_parameters);
-            auto& target_location = find_location(target_type, *person);
-            auto current_location = person->get_location();
-            if (m_testing_strategy.run_strategy(*person, target_location, t)) {
-                if (target_location != current_location &&
-                    target_location.get_number_persons() < target_location.get_capacity().persons) {
-                    bool wears_mask = person->apply_mask_intervention(target_location);
-                    if (wears_mask) {
-                        person->migrate_to(target_location);
+            auto target_type = rule.first(*person, t, dt, m_migration_parameters);
+            if (person->get_assigned_location_index(target_type) != INVALID_LOCATION_INDEX) {
+                auto& target_location = find_location(target_type, *person);
+                auto current_location = person->get_location();
+                if (m_testing_strategy.run_strategy(*person, target_location, t)) {
+                    if (target_location != current_location &&
+                        target_location.get_number_persons() < target_location.get_capacity().persons) {
+                        bool wears_mask = person->apply_mask_intervention(target_location);
+                        if (wears_mask) {
+                            m_movement_data.push_back({person->get_person_id(), current_location.get_index(),
+                                                       target_location.get_index(), t, t + dt,
+                                                       mio::abm::TransportMode::Unknown, mio::abm::UnknownActivity,
```

Comment:
TransportMode and Activity will be set later I guess once we have implemented a traffic model, but you could set Activity here (or better in the loggers) based on the type of location they go to.